README.md

```diff
@@ -292,6 +292,37 @@ echo 'a';
 
 See also [example #3](examples).
 
+### addSignal()
+
+**Note: Signaling is only available on Unix-like platform, Windows isn't supported.**
+
+The `addSignal(int $signal, callable $listener): void` method can be used to
+be notified about OS signals. This is useful to catch user interrupt signals or 
+shutdown signals from tools like `supervisor` or `systemd`.
+
+```php
+$listener = function (int $signal) {
+    echo 'Caught user iterrupt signal', PHP_EOL;
+};
+$loop->addSignal(SIGINT, $listener);
+```
+
+See also [example #4](examples).
+
+**Note: A listener can only be added once to the same signal, any attempts to add it 
+more then once will be ignored.**
```

Comment:
No objection to this limitation, but I would suggest making sure this is closer to the verbiage for our stream API (https://github.com/reactphp/event-loop/pull/110/files#diff-04c6e90faac2675aa89e2176d2eec7d8R312). What do you think about this?

Reply:
Updated it, it isn't my strongest skill and I rather not let it block this PR thus I suggest if it needs work I do it in a follow up PR.